test/e2e/framework/test_context.go

```diff
@@ -300,8 +300,8 @@ func RegisterCommonFlags(flags *flag.FlagSet) {
 	flags.IntVar(&TestContext.AllowedNotReadyNodes, "allowed-not-ready-nodes", 0, "If greater than zero, framework will allow for that many non-ready nodes when checking for all ready nodes. If -1, no waiting will be performed for ready nodes or daemonset pods.")
 
 	flags.StringVar(&TestContext.Host, "host", "", fmt.Sprintf("The host, or apiserver, to connect to. Will default to %s if this argument and --kubeconfig are not set.", defaultHost))
-	flags.StringVar(&TestContext.ReportPrefix, "report-prefix", "", "Optional prefix for JUnit XML reports. Default is empty, which doesn't prepend anything to the default name.")
-	flags.StringVar(&TestContext.ReportDir, "report-dir", "", "Path to the directory where the JUnit XML reports should be saved. Default is empty, which doesn't generate these reports.")
+	flags.StringVar(&TestContext.ReportPrefix, "report-prefix", "", "report-prefix is deprecated and will be removed in 1.28. Optional prefix for JUnit XML reports. Default is empty, which doesn't prepend anything to the default name.")
+	flags.StringVar(&TestContext.ReportDir, "report-dir", "", "report-dir is deprecated and will be removed in 1.28. Path to the directory where the JUnit XML reports should be saved. Default is empty, which doesn't generate these reports.")
```

Comment:
So if I understand this commit right, these two flags are still present for backward-compatibility, but won't write a JUnit report when used, right?

Then why keep them if users have to update their invocation to get the behavior that they want? Getting a command line parsing error will be easier to diagnose than silently not writing a JUnit report.

Reply:
FWIW, I agree with the decision to remove the custom JUnit reporter, we just need to consider the impact.

Speaking of "removing" - where does the code for the reporter get removed? It's not in this commit.

Reply:
If we keep the flags, removing them should be added to the list in https://github.com/kubernetes/kubernetes/issues/109744 and that issue should be referenced here.

Reply:
> Getting a command line parsing error will be easier to diagnose

the parameters are still needed for other purposes, mostly used for saving some files, e.g. `image.json`.  Instead of `paring` error, it's better to update the help message to say this is not used for `junit` reporter anymore, so that we will give some time for end-user who is still use the directory for achieving some logs. 

> Speaking of "removing" - where does the code for the reporter get removed? It's not in this commit.
fair enough, the logic of the `removal` is actually done in the `Ginkgo`, what's the removal here in k8s is removing the custom reporter which was appended to the custom reporter list.

> If we keep the flags, removing them should be added to the list in https://github.com/kubernetes/kubernetes/issues/109744

Done